services/mqtt/client.go

```diff
@@ -1,71 +1,63 @@
 package mqtt
 
 import (
-	"errors"
-	"log"
 	"time"
 
 	pahomqtt "github.com/eclipse/paho.mqtt.golang"
+	"github.com/influxdata/kapacitor/tlsconfig"
 )
 
 // Client describes an immutable MQTT client, designed to accommodate the
 // incongruencies between real clients and mock clients.
 type Client interface {
 	Connect() error
 	Disconnect()
-	Publish(string, byte, bool, string) error
+	Publish(topic string, qos QoSLevel, retained bool, message []byte) error
```

Comment:
It just documents the arguments better. The implementation can still use whatever names it wants.